provider/cluster/manager.go

```diff
@@ -126,17 +129,41 @@ loop:
 			}
 			switch dm.state {
 			case dsDeployActive:
+				if result != nil {
+					break loop
+				}
 				dm.log.Debug("deploy complete")
 				dm.state = dsDeployComplete
 				dm.startMonitor()
 			case dsDeployPending:
+				if result != nil {
+					break loop
+				}
 				// start update
 				runch = dm.startDeploy()
 			case dsDeployComplete:
 				panic(fmt.Sprintf("INVALID STATE: runch read on %v", dm.state))
 			case dsTeardownActive:
-				dm.state = dsTeardownComplete
-				break loop
+				// Teardown completed OK
+				if result == nil {
+					dm.state = dsTeardownComplete
+					break loop
+				}
+
+				// Check to see if the limit on the number of attepmts has been reached
+				if teardownAttempts == maxTeardownAttempts {
+					dm.log.Error("could not teardown lease")
+					break loop
+				}
+
+				// Wait before trying again
+				delay := time.Second * time.Duration(1+math.Exp2(float64(teardownAttempts)))
+				dm.log.Info("Trying teardown again", "delay", delay)
+				time.Sleep(delay)
```

Comment:
We can't `Sleep` in any of these state machine loops.  Can you separate this retry change into a different PR?

Reply:
hmm, where you _can_ sleep is in the routines that are run asynchronously: https://github.com/ovrclk/akash/blob/b0c296d802cd4969c02efb7042edaaf546398d5e/provider/cluster/manager.go#L197-L205

Reply:
This has been removed from this PR